### DISCUSSION TITLE
"Hugging face API" settings

### DISCUSSION THREAD

**Discussion by danielvanmil** [langflow-ai/langflow#1743] [Q&A]
Hello,

I've added a simple test project on a langflow hosted in a Hugging Space space.
Now I want to invoke the inference API of Hugging face in the project.

I added a "Hugging face API" component with endpoint URL "https://api-inference.huggingface.co/models/mistralai/Mistral-7B-Instruct-v0.2", my API key and as model "Mistral-7B-Instruct-v0.2", but I keep getting the error "Error building vertex Hugging Face API: Could not connect to HuggingFace Endpoints API."

Anyone who can help me with the correct settings to make this work or a working example (screenshot)?

**Comment by ogabrielluiz** [comment]
Hey! 
Could it be that you don't have set the HuggingFace API token?

**Reply by fmarangoni** [reply]
I got the same problem, did you find the solution?

**Reply by danielvanmil** [reply]
nope

**Reply by prateekvyas1996** [reply]
same problem, I install transformers because error says no module transformers, in environment and it says huggingfacehub 20 is required I do that also now api issue still there with mistral

**Reply by Lautaro-De-Lucia** [reply]
Same problem here

**Comment by prateekvyas1996** [comment]
Got error "Error Building Component
    Error building vertex Hugging Face API: Failed to resolve model_id:Could not find model id for inference server: https://api-inference.huggingface.co/models/mistralai/Mistral-7B-Instruct-v0.2Make sure that your Hugging Face token has access to the endpoint."

I put following arguments
Endpoint url: https://api-inference.huggingface.co/models/mistralai/Mistral-7B-Instruct-v0.2
Task - text-generation
API token: <My token freshly generated>
Still getting errors.

**Reply by prateekvyas1996** [reply]
I tried to copy the endpoint in brouser url I think I am getting json response from there here is the screenshot
![image](https://github.com/langflow-ai/langflow/assets/28763364/e561b163-e2e3-45e5-9584-5aa56620f4ef)
